### PR TITLE
Feature/api client users

### PR DESCRIPTION
**Pull request by tora-kozic**
### Description of Change ###

Switches the `users` client over to API client compatible endpoints.  Additional lookups are used to preserve backwards compatibility.

### PR Checklist ###
Did you remember to do the below?

- [x] Add unit tests to verify this change
- [x] Add an entry to CHANGELOG.md describing this change
- [x] Add docstrings for any new public parameters / methods / classes

### CODE REVIEW

File: tests/services/test_users.py
Comment (tora-kozic) [inline tests/services/test_users.py:502]: Just found out about the `patch.object` decorator/method!  Very helpful for patching out a helper method that makes an api call

File: tests/services/test_users.py
Comment (tora-kozic) [inline tests/services/test_users.py:310]: opted to unit test the helper methods rather than the full `add/remove_role` functions due to all the additional lookups necessary

**Comment by tora-kozic** [comment]
- [ ] Integration tests failing until mock-servers are updated

File: src/py42/services/users.py
Comment (timabrmsn) [inline src/py42/services/users.py:255]: the query params for this and reactivate got changed on the new apis to just `block` and `unblock` instead of `blockUser`/`unblockUser`

Comment (tora-kozic) [reply src/py42/services/users.py:255]: Oh cool, they must be backwards compatible or something because they were working as is, but I'll update it to the new values 👍

Comment (timabrmsn) [reply src/py42/services/users.py]: It wouldn't error, but the incorrect params would just be ignored. If you tried `deactivate(user_id, block_user=False)` (opposite of default), the user would still be blocked on deactivation.

Comment (tora-kozic) [reply src/py42/services/users.py]: Ohh I see, I must've been testing it against the default values without knowing so the output looked right 🧐

File: src/py42/services/users.py
Comment (timabrmsn) [reply src/py42/services/users.py:418]: Since we're already doing the lookup either way, we could just accept role-id _or_ role-name values and use them if they're valid, just make the check `if (role["roleName"] == role_name) or (role["roleId"] == role_name):`

Then just update the docs like:

```
role_name (str): The name or id of the role to assign to the user. e.g. "Desktop User" (name) or "desktop-user" (id)
```

This keeps it backwards compatible, but makes it easy to re-use the role-id values. I was testing these changes in the CLI and noticed the CLI only returns `role-id` in user results, but requires `role-name` in the arguments, making this change would make that awkward inconsistency go away.

**Review by timabrmsn — APPROVED** [review]

**Pull request merged**
